src/gebt_poc/gen_alpha_2D.cpp

```diff
@@ -99,30 +121,36 @@ std::tuple<State, Kokkos::View<double*>> GeneralizedAlphaTimeIntegrator::AlphaSt
         "algorithm\n"
     );
 
-    for (size_t node = 0; node < n_nodes; node++) {
-        // Perform the linear update part of the generalized alpha algorithm
-        // Algorithm from Table 1, Brüls, Cardona, and Arnold 2012
-        Kokkos::parallel_for(
-            kNumberOfLieAlgebraComponents,
-            KOKKOS_LAMBDA(const size_t i) {
+    using member_type = Kokkos::TeamPolicy<>::member_type;
+    auto node_team_policy = Kokkos::TeamPolicy<>(n_nodes, Kokkos::AUTO(), Kokkos::AUTO());
+    Kokkos::parallel_for(
+        node_team_policy,
+        KOKKOS_LAMBDA(const member_type& member) {
+            auto node = member.league_rank();
+            constexpr auto components = kNumberOfLieAlgebraComponents;
+            Kokkos::parallel_for(Kokkos::TeamVectorRange(member, components), [=](std::size_t i) {
                 algo_acceleration_next(node, i) = (kAlphaFLocal * acceleration(node, i) -
                                                    kAlphaMLocal * algo_acceleration(node, i)) /
                                                   (1. - kAlphaMLocal);
-
+            });
+            Kokkos::parallel_for(Kokkos::TeamVectorRange(member, components), [=](std::size_t i) {
                 delta_gen_coords(node, i) = velocity(node, i) +
                                             h * (0.5 - kBetaLocal) * algo_acceleration(node, i) +
                                             h * kBetaLocal * algo_acceleration_next(node, i);
-
+            });
+            Kokkos::parallel_for(Kokkos::TeamVectorRange(member, components), [=](std::size_t i) {
                 velocity_next(node, i) = velocity(node, i) +
                                          h * (1 - kGammaLocal) * algo_acceleration(node, i) +
                                          h * kGammaLocal * algo_acceleration_next(node, i);
-
+            });
```

Comment:
Is there a guarantee that these inner parallel_fors are not executing concurrently? I.e., does a parallel_for fence before returning? If they can start before the previous one has finished, it could create some difficult-to-find correctness issues since arrays like `algo_acceleration` get read in one loop and then written in the other.

Reply:
I did run the unit tests a bunch using CUDA and didn't run into any problems. But that might have just been Christmas magic saving me.  I'll add some barriers here on Monday.